src/Storages/MergeTree/MergeTreeWhereOptimizer.cpp

```diff
@@ -327,8 +357,14 @@ void MergeTreeWhereOptimizer::analyzeImpl(Conditions & res, const RPNBuilderTree
         {
             /// Consider all conditions good with this setting enabled.
             cond.good = cond.viable;
-            /// Find min position in PK of any column that is used in this condition.
+
             cond.min_position_in_primary_key = findMinPosition(cond.table_columns, primary_key_names_positions);
+            if (!may_use_primary_index && cond.min_position_in_primary_key != std::numeric_limits<Int64>::max() - 1)
```

Comment:
the condition `&& cond.min_position_in_primary_key != std::numeric_limits<Int64>::max() - 1` is redundant